### PR TITLE
Use TargetFramework conditions consistently in libraries

### PR DESCRIPTION
**Pull request by ViktorHofer**
See the docs commit for the desired behavior: https://github.com/dotnet/runtime/commit/b0b09c67d1e24e687dec395c6a71647c361cf1e5.

Additional cleanup:
- Remove unnecessary properties (`Rootnamespace`, `AssemblyName`)
- Move TargetFramework derived properties (or ones that will dervice from TargetFramework at some point) into a/the second property group.

cc @ericstj

### CODE REVIEW

File: docs/coding-guidelines/project-guidelines.md [hunk 1]
```diff
@@ -97,6 +97,44 @@ When building an individual project the `BuildTargetFramework` and `TargetOS` wi
 - .NET Framework latest -> `$(NetFrameworkCurrent)-Windows_NT`
 
 # Library project guidelines
+
+## TargetFramework conditions
```
Comment (ericstj) [inline docs/coding-guidelines/project-guidelines.md:101]: Should we call out that `$(TargetFramework)` should be avoided in PropertyGroup conditions (unless those are in a second PropertyGroup)?

Comment (ViktorHofer) [reply docs/coding-guidelines/project-guidelines.md:101]: Thanks, done :)

**Comment by ViktorHofer** [comment]
Feedback is addressed. Someone pls approve :D

File: src/libraries/Microsoft.Extensions.DependencyInjection/src/Microsoft.Extensions.DependencyInjection.csproj [hunk 2]
```diff
@@ -3,17 +3,15 @@
...
     <DefineConstants Condition="'$(ILEmitBackend)' == 'True'">$(DefineConstants);IL_EMIT</DefineConstants>
-
     <!-- Debug IL generation -->
     <ILEmitBackendSaveAssemblies>False</ILEmitBackendSaveAssemblies>
   </PropertyGroup>
   
   <!-- DesignTimeBuild requires all the TargetFramework Derived Properties to not be present in the first property group. -->
   <PropertyGroup>
-    <DefineConstants Condition="$(TargetFramework.StartsWith('net4')) AND '$(ILEmitBackendSaveAssemblies)' == 'True'">$(DefineConstants);SAVE_ASSEMBLIES</DefineConstants>  
+    <ILEmitBackend Condition="'$(TargetFramework)' != 'netstandard2.0'">True</ILEmitBackend>
```
Comment (safern) [inline src/libraries/Microsoft.Extensions.DependencyInjection/src/Microsoft.Extensions.DependencyInjection.csproj:13]: Should the other define constants be moved down? It depends on this property.

File: src/libraries/System.IO.Packaging/ref/System.IO.Packaging.csproj [hunk 3]
```diff
@@ -1,11 +1,11 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <DefineConstants Condition="'$(TargetFramework)' == 'netstandard1.3'">$(DefineConstants);netcoreapp</DefineConstants>
     <TargetFrameworks>netstandard2.0;netstandard1.3;net46;$(NetFrameworkCurrent)</TargetFrameworks>
     <ExcludeCurrentFullFrameworkFromPackage>true</ExcludeCurrentFullFrameworkFromPackage>
   </PropertyGroup>
   <!-- DesignTimeBuild requires all the TargetFramework Derived Properties to not be present in the first property group. -->
   <PropertyGroup>
+    <DefineConstants Condition="'$(TargetFramework)' == 'netstandard1.3'">$(DefineConstants);netcoreapp</DefineConstants>
```
Comment (Anipik) [inline src/libraries/System.IO.Packaging/ref/System.IO.Packaging.csproj:8]: curious why netcoreapp is getting defined here .

Comment (ViktorHofer) [reply src/libraries/System.IO.Packaging/ref/System.IO.Packaging.csproj:8]: That's just an internal define in the codebase which wasn't cleaned up.

File: src/libraries/Microsoft.Extensions.DependencyInjection/src/Microsoft.Extensions.DependencyInjection.csproj (hunk 2 above)
Comment (ViktorHofer) [reply src/libraries/Microsoft.Extensions.DependencyInjection/src/Microsoft.Extensions.DependencyInjection.csproj:13]: No, it depends on `ILEmitBackendSaveAssemblies` not `ILEmitBackend`.

Comment (safern) [reply src/libraries/Microsoft.Extensions.DependencyInjection/src/Microsoft.Extensions.DependencyInjection.csproj:13]: It depends on `ILEmitBackend` 😄 : 
```xml
<DefineConstants Condition="'$(ILEmitBackend)' == 'True'">$(DefineConstants);IL_EMIT</DefineConstants>
```

File: docs/coding-guidelines/project-guidelines.md (hunk 1 above)
Comment (ericstj) [reply docs/coding-guidelines/project-guidelines.md:101]: Do you know if we're going to validate that?  I recall at one point we talked about it.

File: src/libraries/Microsoft.Extensions.DependencyInjection/src/Microsoft.Extensions.DependencyInjection.csproj (hunk 2 above)
Comment (ViktorHofer) [reply src/libraries/Microsoft.Extensions.DependencyInjection/src/Microsoft.Extensions.DependencyInjection.csproj:13]: oh now I get what you are saying 😁 thanks.

File: docs/coding-guidelines/project-guidelines.md (hunk 1 above)
Comment (ViktorHofer) [reply docs/coding-guidelines/project-guidelines.md:101]: I would prefer to not add infra for that and instead push the project system / msbuild team to make progress on the actual issue.

**Comment by ViktorHofer** [comment]
Ping reviewers, please approve :)

**Review by Anipik — APPROVED** [review]

**Review by ericstj — APPROVED** [review]

**Review by safern — APPROVED** [review]

**Pull request merged**